extensions/eclipse-che-theia-remote-impl-che-server/src/node/che-server-http-service-impl.ts

```diff
@@ -0,0 +1,136 @@
+/**********************************************************************
+ * Copyright (c) 2021 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ***********************************************************************/
+
+import * as https from 'https';
+import * as tunnel from 'tunnel';
+import * as url from 'url';
+
+import axios, { AxiosInstance, AxiosRequestConfig } from 'axios';
+import { inject, injectable } from 'inversify';
+
+import { CertificateService } from '@eclipse-che/theia-remote-api/lib/common/certificate-service';
+import { HttpService } from '@eclipse-che/theia-remote-api/lib/common/http-service';
+
+@injectable()
+export class CheServerHttpServiceImpl implements HttpService {
+  @inject(CertificateService)
+  private certificateService: CertificateService;
+
+  async get(uri: string): Promise<string | undefined> {
+    const axiosInstance = await this.getAxiosInstance(uri);
+    try {
+      const response = await axiosInstance.get(uri, {
+        transformResponse: [data => data],
+        responseType: 'text',
+      });
+      return response.data;
+    } catch (error) {
+      // not found then we return undefined
+      if (error.response && error.response.status === 404) {
+        return undefined;
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * Use proxy and/or certificates.
+   */
+  private async getAxiosInstance(uri: string): Promise<AxiosInstance> {
+    const certificateAuthority = await this.certificateService.getCertificateAuthority();
+
```

Comment:
we definitely need some shared component for that
https://github.com/eclipse-che/che-theia/blob/master/extensions/eclipse-che-theia-plugin-ext/src/node/che-plugin-service.ts#L129
maybe not in this PR

Reply:
yes I want to migrate to this one after